### DISCUSSION TITLE
Cannot heal my network since major upgrade

### DISCUSSION THREAD

**Discussion by 0rsa** [zwave-js/zwave-js#3319] [Q&A] [answered]
Hello,
Using:
node-red-contrib-zwave-js 5.1.2
zwave-js 8.2.3
Aeon Z-Stick 7

I cannot heal my network anymore.
All my nodes are working, there is no problem to control these.

But when I try to heal my network, I got this error for all my nodes:

  "[Node 007] refreshing neighbor list failed: Timeout while waiting for a callback from the controller"
From 1st to 5th attempt, it fails.

Maybe do you know what I'm missing ? It was working fine using node-red-contrib-zwave-js < 5

Thank you

![failed](https://user-images.githubusercontent.com/769750/132876499-d7b3b6e6-eab2-4cbb-9d6b-5022818c42fa.png)
![network](https://user-images.githubusercontent.com/769750/132879487-259193ef-439f-4964-b52b-2d15aca7829d.png)

**Comment by AlCalzone** [comment, marked as answer]
Without a full driver log I can just guess, but some users reported problems with controller firmware versions prior to 7.15.
I'd suggest updating the stick. You can do that with Silab's PC controller utility.

Before you do that, restart for good meaure (and re-plug the stick while shut down).

**Comment by 0rsa** [comment]
Thank you AlCalzone
It was my idea yesterday but I failed to find the right procedure to do it.
I already have Simplcity Studio 5  with Z-Wave 7.16.2.0 SDK installed, key point is now to know how to upgrade stick, there is no documentation about it in Aeotec documentation.
I opened a ticket to aeotec to get some help.
I you have a link describing how to do, it will be welcome :)

**Comment by 0rsa** [comment]
Browsing other discussions, I found the procedure, thanks to @Hedda 
Firmwares are located here by default: C:\SiliconLabs\SimplicityStudio\v5\developer\sdks\gecko_sdk_suite\v3.2\protocol\z-wave\Apps\bin\gbl
Launch Simplcity Studio, click on Tools and select Z-Wave PC Controller
Then click on OTW Firmware file and select for example ZW_SerialAPI_Controller_7.16.2_267_EFR32ZG23_REGION_EU.gbl if you are in Europe.
Firmware update starts after file selection.

I'm now with 7.16 and healing is re-working!

Thanks!!!

![716](https://user-images.githubusercontent.com/769750/132934424-a7f04264-95b1-4808-a8e5-24e0900de021.png)

**Reply by kpine** [reply]
I am curious, why did you choose the EFR32ZG23 firmware file instead of the EFR32ZG14 one? Does your Z-Stick use the EFR32ZG23? I had thought all of the 700-series USB controllers were using the same SoC (EFR32ZG14?). I am not finding much information about the EFR32ZG23.

**Reply by Hedda** [reply]
I too thought that all Z-Wave 700 USB controller adapters was based on the EFR32ZG14 chip?

Same as the reference hardware -> https://github.com/zwave-js/node-zwave-js/discussions/2544

Aeotec Z-Stick 7 is however listed as based on ZGM130 but then the question is if the CPU and storage is the same

https://help.aeotec.com/support/solutions/articles/6000230554-z-stick-7-technical-specifications-

Aeotec Z-Pi 7 is at least listed as based on EFR32ZG14 

https://aeotec.com/z-wave-home-automation/development-kit-pcb.html

**Reply by 0rsa** [reply]
Random choice to be honest.
Should I flash EFR32ZG14?
Aetoc support answered me and they attached EFR32ZG14 firmware

**Reply by Hedda** [reply]
If the dongle manufacturer provides their own firmware file then you should really use it.

Otherwise:

EFR32ZG14 based dongle should use ZW_SerialAPI_Controller_7.16.2_267_EFR32ZG14_REGION_EU.gbl

ZGM130S based dongle should use ZW_SerialAPI_Controller_7.16.2_267_ZGM130S_REGION_EU.gbl

**Reply by 0rsa** [reply]
Thanks, I will do it.
FYI Aetoc provides me precisely ZW_SerialAPI_Controller_7.15.4_256_EFR32ZG14_REGION_EU.gbl

**Reply by 0rsa** [reply]
Using Aeotec firmware or ZW_SerialAPI_Controller_7.16.2_267_EFR32ZG14_REGION_EU.gbl, it fails now.
![update](https://user-images.githubusercontent.com/769750/133247970-3aa85ff2-4b68-4577-90ac-a3348a46e913.png)
Fortunately it fails before starting so my stick is still working with ZW_SerialAPI_Controller_7.16.2_267_EFR32ZG23_REGION_EU.gbl

**Reply by blhoward2** [reply]
I’ve confirmed with Aeotec that the G14 is the correct file. The G23 is a second generation chip and while the file may work, its not intended for the chip being used.

**Reply by Hedda** [reply]
I think my own upgrade of Silabs UZB7 USB stick initially failed because I first forgot to disable Energy Saving Mode on the device in Windows:

https://silabs-prod.adobecqms.net/community/wireless/z-wave/knowledge-base.entry.html/2020/06/04/uzb7_not_accessibleonwindows10-0Dw1

**Reply by 0rsa** [reply]
According to Aetoec support, you can't flash a firmware if you already have a newer or the same firmware, so that's why I have this error message.
I have to wait for a new Z-Wave release version to upgrade to the right chipset (EFR32ZG14).
It is a bit annoying so be careful when flashing a firmware. I don't  know why Aetoec doesn't provide an online documentation to do that, opening a ticket for an upgrade is a bit "overkill".

**Reply by blhoward2** [reply]
It's not their file or their process. It's SiLabs.

**Reply by Hedda** [reply]
> It's not their file or their process. It's SiLabs.

I think that he means that Aeotec could provide the GBL firmware binary files downloads via their website without having to open a support case as well and make a basic script/tool for simple OTW (Over The Wire) upgrades. See ex. "Elelabs EZSP Firmware Update Utility" Bootloader XModem flash -> https://github.com/Elelabs/elelabs-zigbee-ezsp-utility and https://github.com/walthowd/husbzb-firmware

**Reply by blhoward2** [reply]
Last I spoke to Aeotec about that, they weren't even sure they could distribute the GBL without telling people to download Simplicity Studio. I believe SiLabs gave them permission to do that now, but not to post it publicly.

**Reply by Hedda** [reply]
As I understand, their license allows you to distribute binaries, but only to specifically be used on the Silabs hardware that it is built for.

**Reply by 0rsa** [reply]
Aeotec posted a documentation here:
https://help.aeotec.com/support/solutions/articles/6000252296-update-z-stick-7-sdk-7-15
